var/spack/repos/builtin/packages/netlib-lapack/package.py

```diff
@@ -149,13 +150,16 @@ def install_one(self, spec, prefix, shared):
             cmake_args.extend(['-DCBLAS=OFF'])
             cmake_args.extend(['-DLAPACKE:BOOL=OFF'])
 
-        if self.compiler.name == 'xl' or self.compiler.name == 'xl_r':
-            # use F77 compiler if IBM XL
-            cmake_args.extend([
-                '-DCMAKE_Fortran_COMPILER=%s' % self.compiler.f77,
-                '-DCMAKE_Fortran_FLAGS=%s' % (
-                    ' '.join(self.spec.compiler_flags['fflags'])),
-            ])
+        if spec.satisfies('arch=linux-rhel7-ppc64le'):
+            if (self.compiler.name == 'xl' or
+                self.compiler.name == 'xl_r' or
+                self.compiler.name == 'clang'):
+                # use F77 compiler if IBM XL or clang
+                cmake_args.extend([
+                    '-DCMAKE_Fortran_COMPILER=%s' % self.compiler.f77,
```

Comment:
The reason for imposing the compiler is that on CORAL we'll use clang+xlf and not clang+flang; also, that was the reason I forced it only for linux-rhel7-ppc64le architecture.

I think it's better to add the xlf compiler in compilers/clang.py, but don't know how to select the patform for ppc64le. For example, in that file, there is the following code:

    ```
     55     if sys.platform == 'darwin':
     56         # Use default wrappers for fortran, in case provided in
     57         # compilers.yaml
     58         link_paths['f77'] = 'clang/gfortran'
     59         link_paths['fc'] = 'clang/gfortran'
     60     else:
     61         link_paths['f77'] = 'clang/flang'
     62         link_paths['fc'] = 'clang/flang'
```

That's where I wanted to add the link_paths for f77 and fc to be clang/xl_r, but how do I select only for linux-rhel7-ppc64le?  sys.platform is set to 'linux2' on my test system.  Is that specific to this platform?--I don't have Linux on Intel.